fileghost/fileghost.py

```diff
@@ -1,114 +1,139 @@
-import secrets
 import json
-import sys
 import os
-global SEP
-SEP='__FIL3GH0ST__'
-global SEPARATOR
-SEPARATOR=list(bytes(SEP.encode()))
-
-def keystore_validity_test(keys:dict):
-    def valid_values():
-        flag=True
-        for v in keys.values():
-            if v>255:
-               flag=False
-               return flag
-               break
-        return flag
-    print("Checking for keystore validity...")
-    assert isinstance(keys,dict), "Keystore must be a dictionary"
-    assert len(keys)==256, "Keystore keys count must be 256, but "+str(len(keys))+" instead"
-    assert sum(keys.keys())==32640, "Sum of keystore keys should be 32640, but "+str(sum(keys.keys()))+" instead"
-    assert len(keys.values())==len(set(keys.values())), "Keystore values must be unique and must not repeat, but duplicates found"
-    assert valid_values()==True, "Any keystore values must be less than 256"
-    
-    return True
-class keygen():
-    def __init__(self):
-        global l
-        l=[]
-        while len(l) != 256:
-              r=secrets.randbelow(256)
-              if r not in l:
-                 l.append(r)
-    def to_keystore(self):#needed for file encryption/decryption
-        s = set(range(256))
-        keystore=dict.fromkeys(s)
-        for x in keystore:
-            keystore[x]=l[x]
-        return keystore
-        return hl
-
-    def to_byte_array(self):
-        return l    
+import secrets
+import sys
+
+
+SEP = '__FIL3GH0ST__'.encode()
+SEPARATOR = list(bytes(SEP))
+
+
+class keygen:
+    @classmethod
+    def generate(cls) -> "keygen":
+        '''
+        Generates a new keygen object with random keys, based on the ``secrets`` module.
+        '''
+        keys = []
+
+        numbers = set()
+        while len(numbers) < 256:
+            n = secrets.randbelow(256)
+
+            if n not in numbers:
+                numbers.add(n)
+                keys.append(n)
+
+        return cls(keys)
+
+    @classmethod
+    def from_file(cls, path: str) -> "keygen":
+        '''
+        Loads a keygen object from a file.
+        '''
+        with open(path, "r") as f:
+            keys = json.loads(f.read())
+
+        return cls(list(keys.values()))
+
+    def __init__(self, keys: list) -> None:
+        if set(keys) != set(range(256)):
+            raise ValueError("Keystore must contain all numbers from 0 to 255")
+
+        self._keys = keys
+        self._keystore = dict(enumerate(self._keys))
+
+    def to_keystore(self) -> dict:
+        return self._keystore.copy()
+
+    def to_byte_array(self) -> list:
+        return self._keys.copy()
```

Comment:
This returns a copy, since dictionaries (and lists) are mutable in python, which means that the user could accidentally change the keys on the instance, and thus ruin their ability to encrypt/decrypt. In general, it's good for public facing interfaces to make defensive copies of mutuable objects that they hand out